compatibility/mysql/csharp/MySQLTest.cs

```diff
@@ -0,0 +1,179 @@
+
+        public void Connect(string ip, int port, string user, string password)
+        {
+            try
+            {
+                string connectionString = $"Server={ip};Port={port};User Id={user};Password={password};";
+                conn = new MySqlConnection(connectionString);
+                conn.Open();
+                cmd = conn.CreateCommand();
+                cmd.CommandType = CommandType.Text;
+            }
+            catch (MySqlException e)
+            {
+                throw new Exception(e.Message);
+            }
+        }
+
+        public void Disconnect()
+        {
+            try
+            {
+                cmd.Dispose();
+                conn.Close();
+            }
+            catch (MySqlException e)
+            {
+                throw new Exception(e.Message);
+            }
+        }
+
+        public void AddTest(string query, string[][] expectedResults)
+        {
+            tests.Add(new Test(query, expectedResults));
+        }
+
+        public bool RunTests()
+        {
+            foreach (var test in tests)
+            {
+                if (!test.Run(cmd))
+                {
+                    return false;
+                }
+            }
+            return true;
+        }
+
+        public void ReadTestsFromFile(string filename)
+        {
+            try
+            {
+                using (var reader = new StreamReader(filename))
+                {
+                    string line;
+                    while ((line = reader.ReadLine()) != null)
+                    {
+                        if (string.IsNullOrWhiteSpace(line)) continue;
+                        string query = line;
+                        var results = new List<string[]>();
+                        while ((line = reader.ReadLine()) != null && !string.IsNullOrWhiteSpace(line))
+                        {
+                            results.Add(line.Split(','));
+                        }
+                        string[][] expectedResults = results.ToArray();
+                        AddTest(query, expectedResults);
+                    }
+                }
+            }
+            catch (IOException e)
+            {
+                Console.Error.WriteLine(e.Message);
+                Environment.Exit(1);
+            }
+        }
+
+        public class Test
+        {
+            private string query;
+            private string[][] expectedResults;
+
+            public Test(string query, string[][] expectedResults)
+            {
+                this.query = query;
+                this.expectedResults = expectedResults;
+            }
+
+            public bool Run(MySqlCommand cmd)
+            {
+                try
+                {
+                    Console.WriteLine("Running test: " + query);
+                    cmd.CommandText = query;
+                    using (var reader = cmd.ExecuteReader())
+                    {
+                        if (!reader.HasRows)
```

Comment:
It's better to move result validation logic to a separate method `ValidateResults`.

Reply:
Got it, let's keep this one as it is.